hazelcast/src/main/java/com/hazelcast/jmx/MapMBean.java

```diff
@@ -74,6 +80,8 @@ public void mapCleared(MapEvent event) {
             }
         };
         listenerId = managedObject.addEntryListener(entryListener, false);
+        StatSupplier<LocalMapStats> localMapStatsSupplier = new LocalMapStatsSupplier(managedObject);
```

Comment:
It might be make it more complicated. Will it hold a reference to itself that way?

`this.localMapStatsDelegate = new LocalStatsDelegate<LocalMapStats>(**this**, updateInterval);`